src/systems/ec/implementations/react/components/site-header-core/src/SiteHeaderCore.jsx

```diff
@@ -52,6 +56,53 @@ const SiteHeaderCore = ({
             />
           </a>
           <div className="ecl-site-header-core__action">
+            {!!(
+              loginToggle &&
+              Object.values(loginToggle).length >= 1 &&
+              loginBox
```

Comment:
Not as easy to follow, and a bit tricky with the "bang-bang" (`!!`). I think it's ok to skip the "truthy" typecasting at all or use Boolean if required.

The check for number of items in `loginToggle` is brittle.Two properties will make it pass, but won't be sufficient for later use. Similar concern for the `loginBox` object which can easily be truthy without having sufficient properties to be useful.

I think a longer and more verbose check can be ok and if any of the properties is missing it'd be better to skip displaying the login part at all.

Reply:
ideally, every properties should be checked, indeed.
But as react packages are not released (they all are set to private), it is only for internal use. So we kind of control the data the pass.
So I'd consider it as a nice to have, when we have time (there are probably lots of similar cases in other components)